src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/X509Certificates/StorePal.Android.AndroidKeyStore.cs

```diff
@@ -118,7 +118,7 @@ public void CloneTo(X509Certificate2Collection collection)
 
             private static string GetCertificateHashString(ICertificatePal certPal)
             {
-                return X509Certificate.GetCertHashString(HashAlgorithmName.SHA256, certPal);
+                return X509Certificate.GetCertHashString(HashAlgorithmName.SHA256, certPal.RawData);
```

Comment:
Does the Android PAL memoize that?  Or is it moot because this is a transitory PAL object anyways?

Reply:
> Or is it moot because this is a transitory PAL object anyways?

That. We don't have a "live" X.509 certificate in this call chain, only the PAL.